File: backend/app/schemas/briefing.py
import uuid
from datetime import datetime, date
from typing import List

from pydantic import BaseModel


class BriefingOut(BaseModel):
    id: uuid.UUID
    date: date
    top_themes: List
    deep_dives: List
    generated_at: datetime

    model_config = {"from_attributes": True}


class PaginatedBriefings(BaseModel):
    total: int
    page: int
    page_size: int
    items: List[BriefingOut]

File: frontend/src/types/briefing.ts
export interface Theme {
  rank: number;
  title: string;
  description: string;
  tags: string[];
  sources: string[];
}

export interface DeepDive {
  title: string;
  body: string;
  sources: string[];
}

export interface Briefing {
  id: string;
  date: string;
  top_themes: Theme[];
  deep_dives: DeepDive[];
  generated_at: string;
}

export interface PaginatedBriefings {
  total: number;
  page: number;
  page_size: number;
  items: Briefing[];
}
